src/main/scala/com/gu/subscriptions/cas/directives/ResponseCodeTransformer.scala

```diff
@@ -0,0 +1,43 @@
+package com.gu.subscriptions.cas.directives
+
+import com.gu.cas.CAS.CASError
+import com.gu.subscriptions.cas.model.json.ModelJsonProtocol._
+import spray.http.HttpResponse
+import spray.http.StatusCodes._
+import spray.json._
+
+import scala.util.{Failure, Success, Try}
+
+object ResponseCodeTransformer {
+  /**
+   * Change the response code that is returned by the call to CAS.
+   * This is necessary because in some cases CAS sends a 200 response
+   * containing an error, making the API misleading.
+   * The error codes are defined there:
+   * https://github.com/guardian/content-authorisation/blob/master/content-authorisation/src/main/resources/conf/global.properties
+   */
+
+  val badRequestCodes = Seq(-40, -50, -80)
+  case class BadRequestCode(code: Int)
+  object BadRequestCode {
+    def unapply(code: Int) = if (badRequestCodes.contains(code)) Some(BadRequestCode(code)) else None
+  }
+
+  val unauthorizedCodes = Seq(-60, -70, -90, -100, -110, -120)
+  case class UnauthorizedCode(code: Int)
+  object UnauthorizedCode {
+    def unapply(code: Int) = if (unauthorizedCodes.contains(code)) Some(UnauthorizedCode(code)) else None
+  }
+
+  val changeResponseCode: HttpResponse => HttpResponse = resp =>
+    Try {
+      resp.entity.asString.parseJson.convertTo[CASError]
+    } match {
+      // Caveat: A malformed json payload send to the /auth endpoint will result in an HTML 404 response by
```

Comment:
is this comment about the failure case? if so, perhaps is better to have it on top the failure.